Solved

Hi, i have annoying problem after install this plugin, i tried to run in my android phone and got error.

with 'import'
`import {Telephony} from "nativescript-telephony"`

result : 

```
app/main-view-model.ts(4,25): error TS2307: Cannot find module 'nativescript-telephony'.
TypeScript compiler failed with exit code 1
```

with 'var'
`var Telephony = require('nativescript-telephony');`

result :

```
TypeError: Telephony is not function
```

I've missed something?

Thank you

--- a/.npmignore
+++ b/.npmignore
@@ -3,4 +3,5 @@ demo/
 *.log
 *.map
 *.ts
-!.d.ts
+!*.d.ts
+*.sublime-project

--- a/demo/package.json
+++ b/demo/package.json
@@ -7,11 +7,14 @@
     "id": "org.nativescript.demo",
     "tns-ios": {
       "version": "2.2.1"
+    },
+    "tns-android": {
+      "version": "2.5.0"
     }
   },
   "dependencies": {
     "nativescript-telephony": "file:..",
-    "tns-core-modules": "2.2.0"
+    "tns-core-modules": "2.5.0"
   },
   "devDependencies": {
     "babel-traverse": "6.7.6",
@@ -21,6 +24,6 @@
     "lazy": "1.0.11",
     "tns-platform-declarations": "^2.2.0",
     "nativescript-dev-typescript": "^0.3.2",
-    "typescript": "^1.8.10"
+    "typescript": "~2.1.0"
   }
 }

--- a/demo/tsconfig.json
+++ b/demo/tsconfig.json
@@ -2,7 +2,6 @@
     "compilerOptions": {
         "module": "commonjs",
         "target": "es5",
-        "sourceMap": true,
         "experimentalDecorators": true,
         "emitDecoratorMetadata": true,
         "noEmitHelpers": true,

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
 	"name": "nativescript-telephony",
 	"version": "1.0.2",
 	"description": "A Nativescript plugin to get the device's SIM data (carrier name, mcc mnc, country code, telephonenumber, etc)",
-	"main": "telephony.js",
+	"main": "telephony",
 	"typings": "index.d.ts",
 	"nativescript": {
 		"platforms": {
@@ -52,7 +52,7 @@
 	},
 	"devDependencies": {
 		"tns-platform-declarations": "^2.0.0",
-		"typescript": "^1.8.10"
+		"typescript": "~2.1.0"
 	}
 }
 

--- a/platforms/android/AndroidManifest.xml
+++ b/platforms/android/AndroidManifest.xml
@@ -1,5 +1,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android">
-    <uses-permission android:name="android.permission.READ_PHONE_STATE" />
-</manifest>
-

--- a/telephony.android.ts
+++ b/telephony.android.ts
@@ -1,15 +1,15 @@
-// 
+//
 
 /**
  *
- * 
-  .oooooo.   ooooooooo.   oooooooooooo oooooooooo.   ooooo ooooooooooooo  .oooooo..o 
- d8P'  `Y8b  `888   `Y88. `888'     `8 `888'   `Y8b  `888' 8'   888   `8 d8P'    `Y8 
-888           888   .d88'  888          888      888  888       888      Y88bo.      
-888           888ooo88P'   888oooo8     888      888  888       888       `"Y8888o.  
-888           888`88b.     888    "     888      888  888       888           `"Y88b 
-`88b    ooo   888  `88b.   888       o  888     d88'  888       888      oo     .d8P 
- `Y8bood8P'  o888o  o888o o888ooooood8 o888bood8P'   o888o     o888o     8""88888P'  
+ *
+  .oooooo.	 ooooooooo.   oooooooooooo oooooooooo.	 ooooo ooooooooooooo  .oooooo..o
+ d8P'  `Y8b  `888	`Y88. `888'		`8 `888'   `Y8b  `888' 8'	888   `8 d8P'	 `Y8
+888			  888	.d88'  888			888		 888  888		888		 Y88bo.
+888			  888ooo88P'   888oooo8		888		 888  888		888		  `"Y8888o.
+888			  888`88b.	   888	  "		888		 888  888		888			  `"Y88b
+`88b	ooo   888  `88b.   888		 o	888		d88'  888		888		 oo		.d8P
+ `Y8bood8P'  o888o	o888o o888ooooood8 o888bood8P'	 o888o	   o888o	 8""88888P'
  *
  *
  * http://www.network-science.de/ascii/
@@ -34,7 +34,7 @@ declare let android: any
 
 function hasPermission(): boolean {
 	// if (android.os.Build.VERSION.SDK_INT < 23) {
-	// 	return true
+	//	   return true
 	// }
 
 	if (!android.support || !android.support.v4 || !android.support.v4.content || !android.support.v4.content.ContextCompat || !android.support.v4.content.ContextCompat.checkSelfPermission) {
@@ -60,14 +60,14 @@ export function Telephony(): Promise<any> {
 
 			let reqid: number = Math.floor(Math.random() * 999)
 
-			function onPermissionsEvent(args) {
+			application.android.addEventListener(application.AndroidApplication.activityRequestPermissionsEvent, function onPermissionsEvent(args) {
 
-				if (args.requestCode == reqid && args.permissions[0] == android.Manifest.permission.READ_PHONE_STATE) {
+				if ((<any>args).requestCode == reqid && (<any>args).permissions[0] == android.Manifest.permission.READ_PHONE_STATE) {
 
 					// removeEventListener to reduce memory usage since it doesnt need to listen anymore
 					application.android.removeEventListener(application.AndroidApplication.activityRequestPermissionsEvent, onPermissionsEvent)
 
-					if (args.grantResults[0] == android.content.pm.PackageManager.PERMISSION_GRANTED) {
+					if ((<any>args).grantResults[0] == android.content.pm.PackageManager.PERMISSION_GRANTED) {
 						resolve(true)
 					} else {
 						resolve(false)
@@ -75,9 +75,8 @@ export function Telephony(): Promise<any> {
 					}
 				}
 
-			}
+			});
 
-			application.android.addEventListener(application.AndroidApplication.activityRequestPermissionsEvent, onPermissionsEvent)
 			android.support.v4.app.ActivityCompat.requestPermissions(application.android.foregroundActivity, [android.Manifest.permission.READ_PHONE_STATE], reqid)
 
 		}

